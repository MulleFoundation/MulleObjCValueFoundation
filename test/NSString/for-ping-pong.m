#ifndef __MULLE_OBJC__
# import <Foundation/Foundation.h>
#else
# import <MulleObjCValueFoundation/MulleObjCValueFoundation.h>
#endif


static void    test( NSString *s)
{
   struct MulleStringEnumerator   rover;
   unichar      c, d;

   _MulleStringEnumeratorInit( &rover, s);

   if( MulleStringEnumeratorNext( &rover, &c))
      mulle_printf( "n:'%lC'\n", c);

   if( MulleStringEnumeratorPrevious( &rover, &c))
      mulle_printf( "p:'%lC'\n", c);

   if( MulleStringEnumeratorPrevious( &rover, &c))
      mulle_printf( "p:'%lC'\n", c);

   if( MulleStringEnumeratorNext( &rover, &c))
      mulle_printf( "n:'%lC'\n", c);

   if( MulleStringEnumeratorNext( &rover, &c))
      mulle_printf( "n:'%lC'\n", c);

   if( MulleStringEnumeratorNext( &rover, &c))
      mulle_printf( "n:'%lC'\n", c);

   if( MulleStringEnumeratorPrevious( &rover, &c))
      mulle_printf( "p:'%lC'\n", c);

   if( MulleStringEnumeratorPrevious( &rover, &c))
      mulle_printf( "p:'%lC'\n", c);

   if( MulleStringEnumeratorPrevious( &rover, &c))
      mulle_printf( "p:'%lC'\n", c);

   MulleStringEnumeratorDone( &rover);
}



int   main( void)
{
   test( @"VfL");

   return( 0);
}
