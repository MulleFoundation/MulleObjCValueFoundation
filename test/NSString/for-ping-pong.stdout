n:'V'
p:'V'
n:'V'
n:'f'
n:'L'
p:'L'
p:'f'
p:'V'
